tests/drivers/uart/uart_async_api/src/test_uart_async.c

```diff
@@ -707,6 +707,7 @@ ZTEST_USER(uart_async_write_abort, test_write_abort)
 
 	uart_tx(uart_dev, tx_buf, 95, 100 * USEC_PER_MSEC);
 	uart_tx_abort(uart_dev);
+	k_sleep(K_MSEC(30));
 	zassert_equal(k_sem_take(&tx_aborted, K_MSEC(100)), 0,
 		      "TX_ABORTED timeout");
 	if (sent != 0) {
```

Comment:
Adding `k_sleep()` here is equivalent to modify timeout on 2 following checks from 100 ms to 70 ms.
Did you try to put the `k_sleep()` just before `zassert_equal(sent, received, "Sent is not equal to received.");` to avoid this ?
Otherwise we might want to increase timeouts from 100 to 130 ms to avoid side effects of the change, but I may be too much picky here.

Reply:
Putting the function `k_sleep()` just before `zassert_equal(sent, received, "Sent is not equal to received.");`  works well.

Reply:
Ok, then let's make the change then. txs